docker/downloader.sh

```diff
@@ -0,0 +1,618 @@
+    break
+    ;;
+  0.2.2)
+    FEDIMINT_VERSION="0.2.2"
+    break
+    ;;
+  0.2.1)
+    FEDIMINT_VERSION="0.2.1"
+    break
+    ;;
+  *)
+    echo "Invalid version. Please run the script again and select a valid version."
+    ;;
+  esac
+done
+
+echo "Checking docker and other required dependencies..."
+
+DOCKER_COMPOSE=docker-compose
+if docker compose version | grep 'Docker Compose' >&/dev/null; then
+  DOCKER_COMPOSE="docker compose"
+elif ! [ -x "$(command -v docker-compose)" ]; then
+  # check if we are running as root
+  if [ "$EUID" -ne 0 ]; then
+    echo 'Error: docker-compose is not installed and we can not install it for you.' >&2
+    exit 1
+  fi
+  if [ -x "$(command -v apt)" ]; then
+    apt install -y docker-compose
+  elif [ -x "$(command -v yum)" ]; then
+    yum install -y docker-compose
+  elif [ -x "$(command -v dnf)" ]; then
+    dnf install -y docker-compose
+  elif [ -x "$(command -v pacman)" ]; then
+    pacman -S --noconfirm docker-compose
+  elif [ -x "$(command -v apk)" ]; then
+    apk add docker-compose
+  else
+    echo 'Error: docker-compose is not installed and we could not install it for you.' >&2
+    exit 1
+  fi
+  if ! [ -x "$(command -v docker-compose)" ]; then
+    echo 'Error: docker-compose is not installed and we could not install it for you.' >&2
+    exit 1
+  fi
+fi
+
+COMMANDS="awk curl sed tr wc jq"
+for command in $COMMANDS; do
+  if ! [ -x "$(command -v $command)" ]; then
+    echo "Error: $command is not installed. Please try to install it" >&2
+    exit 1
+  fi
+done
+
+if [ "$(awk '/MemTotal/ {print $2}' /proc/meminfo)" -lt 900000 ]; then
+  echo 'Error: Your machine must have at least 1GB of RAM' >&2
+  exit 1
+fi
+
+echo "Dependencies are ready. Let's continue..."
+
+while true; do
+  echo
+  echo "Do you want to install a Fedimint Guardian or a Lightning Gateway? [guardian/gateway]"
+  echo "Guardian: connects with other guardians to form and run a Fedimint"
+  echo "Gateway: a lightning network service provider to users of Fedimints"
+  read -p "Type 'guardian' or 'gateway': " INSTALL_TYPE
+
+  case $INSTALL_TYPE in
+  guardian)
+    SERVICES="fedimintd guardian-ui"
+    IS_GATEWAY=false
+    break
+    ;;
+  gateway)
+    SERVICES="gatewayd gateway-ui thunderhub"
+    IS_GATEWAY=true
+    break
+    ;;
+  *)
+    echo "Invalid option. Please type 'guardian' or 'gateway'."
+    ;;
+  esac
+done
+
+while true; do
+  echo
+  echo "Do you want to setup TLS certificates with Let's Encrypt? [yes/no]"
+  read -p "Type 'yes' to setup TLS certificates or 'no' to skip: " SETUP_TLS
+
+  case $SETUP_TLS in
+  yes)
+    SETUP_TLS=true
+    TLS_DIR="tls"
+    break
+    ;;
+  no)
+    SETUP_TLS=false
+    TLS_DIR="no-tls"
+    break
+    ;;
+  *)
+    echo "Invalid option. Please type 'yes' or 'no'."
+    ;;
+  esac
+done
+
+DOCKER_COMPOSE_FILE="https://raw.githubusercontent.com/Kodylow/fedimint/kl/docker-deploy/docker/latest/${INSTALL_TYPE}/${TLS_DIR}/docker-compose.yaml"
```

Comment:
nit: reminder to switch it back to fedimint right before merging

Reply:
tracking ty